internal/loader/typed_setting.go

```diff
@@ -118,3 +116,80 @@ func isJsonContentType(contentType *string) bool {
 	matched, _ := regexp.MatchString("^application\\/(?:[^\\/]+\\+)?json(;.*)?$", contentTypeStr)
 	return matched
 }
+
+func UnmarshalConfigMap(existingConfigMapSetting *map[string]string, dataOptions *acpv1.ConfigMapDataOptions) (interface{}, map[string]interface{}, error) {
+	allSettings := make(map[string]interface{})
+	existingConfigMapData := (*existingConfigMapSetting)[dataOptions.Key]
+	var typedFeatureFlags interface{}
+	var err error
+
+	switch dataOptions.Type {
+	case acpv1.Yaml:
+		err = yaml.Unmarshal([]byte(existingConfigMapData), &allSettings)
+		if err != nil {
+			return nil, nil, err
+		}
+	case acpv1.Json:
+		err = json.Unmarshal([]byte(existingConfigMapData), &allSettings)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	if _, ok := allSettings[FeatureManagementSectionName]; ok {
+		typedFeatureFlags = allSettings[FeatureManagementSectionName]
+		delete(allSettings, FeatureManagementSectionName)
+	}
+
+	return typedFeatureFlags, allSettings, nil
+}
+
+func createJsonYamlTypeData(settings map[string]interface{}, dataOptions *acpv1.ConfigMapDataOptions) (string, error) {
+	switch dataOptions.Type {
+	case acpv1.Yaml:
+		yamlStr, err := yaml.Marshal(settings)
+		if err != nil {
+			return "", fmt.Errorf("Failed to marshal key-values to yaml: %s", err.Error())
+		}
+		return string(yamlStr), nil
+	case acpv1.Json:
+		jsonStr, err := json.Marshal(settings)
+		if err != nil {
+			return "", fmt.Errorf("Failed to marshal key-values to json: %s", err.Error())
+		}
+		return string(jsonStr), nil
+	}
+
+	return "", nil
+}
+
+func unmarshalFeatureFlagSettings(featureFlagSettings interface{}) (interface{}, error) {
+	if featureFlagSettings == nil {
+		return nil, nil
+	}
+
+	switch featureFlagSettings.(type) {
+	// feature flag settings from existing configMap
+	case map[string]interface{}:
+		return featureFlagSettings, nil
+	// feature flag settings from appconfig store
+	case []*string:
```

Comment:
I can't catch up why we need to handle this type. feature flag settings from appconfig store should be also `map[string]interface{}` type

Reply:
`UnmarshalConfigMap` will return typed feature flag settings. When retrieving from appconfig store, feature flag settings are JSON-encoded data which need to call `json.Unmarshal` and store the values in `interface{}` for subsequent `json/yaml` format configMap data generation.